src/match/controllers/create_match_controller.py

```diff
@@ -0,0 +1,46 @@
+from api.utils import failure_response
+from api.utils import success_response
+from django.contrib.auth.models import User
+from match import match_status
+from match.models import Match
+from rest_framework import status
+
+
+class CreateMatchController:
+    def __init__(self, data, serializer):
+        self._data = data
+        self._serializer = serializer
+
+    def process(self):
+        """Process a request to create a match."""
+        match_ids = self._data.get("ids")
+        if match_ids is None:
+            return failure_response(
+                "POST body is misformatted", status.HTTP_400_BAD_REQUEST
+            )
+        user_1 = User.objects.filter(id=match_ids[0])
+        user_2 = User.objects.filter(id=match_ids[1])
+        if not user_1:
+            return failure_response(f"User with id {match_ids[0]} does not exist")
+        if not user_2:
+            return failure_response(f"User with id {match_ids[1]} does not exist")
+        user_1 = user_1[0]
+        user_2 = user_2[0]
+        possible_match_1 = Match.objects.filter(
+            user_1=user_1, user_2=user_2, status=match_status.CREATED
+        )
+        possible_match_2 = Match.objects.filter(
+            user_1=user_2, user_2=user_1, status=match_status.CREATED
+        )
+        if possible_match_1 or possible_match_2:
```

Comment:
The top comment ESPECIALLY applies here because unlike users, we will be getting a linear increase of matches per week so if `possible_match_1` exists we can just return success immediately. 

A nice precondition for a future optimization PR that we can have is if we were to guarantee that `user_1` always has a lower `id` than `user_2` so we only need to filter through the matches a single time before creating a new one.

Also good job filtering by the status too because we want to ensure that even if someone on Pear were to theoretically meet everyone on the app that they can possibly match again with a good Pear in the future after some time period.

Reply:
smart boy